src/linear_algebra/clifford_algebra/conjugation.lean

```diff
@@ -0,0 +1,132 @@
+* `clifford_algebra.reverse_involutive`
+* `clifford_algebra.reverse_involute_commute`
+-/
+
+variables {R : Type*} [comm_ring R]
+variables {M : Type*} [add_comm_group M] [module R M]
+variables {Q : quadratic_form R M}
+
+namespace clifford_algebra
+
+section involute
+
+/-- Grade involution, inverting the sign of each basis vector -/
+def involute : clifford_algebra Q →ₐ[R] clifford_algebra Q :=
+clifford_algebra.lift Q ⟨-(ι Q), λ m, by simp⟩
+
+@[simp] lemma involute_ι (m : M) : involute (ι Q m) = -ι Q m :=
+lift_ι_apply _ _ m
+
+@[simp] lemma involute_comp_involute : involute.comp involute = alg_hom.id R (clifford_algebra Q) :=
+by { ext, simp }
+
+lemma involute_involutive : function.involutive (involute : _ → clifford_algebra Q) :=
+alg_hom.congr_fun involute_comp_involute
+
+end involute
+
+section reverse
+open opposite
+
+/-- Grade reversion, inverting the multiplication order of basis vectors. -/
+def reverse : clifford_algebra Q →ₗ[R] clifford_algebra Q :=
+(op_linear_equiv R).symm.to_linear_map.comp (
+  clifford_algebra.lift Q ⟨(opposite.op_linear_equiv R).to_linear_map.comp (ι Q),
+    λ m, unop_injective $ by simp⟩).to_linear_map
+
+@[simp] lemma reverse_ι (m : M) : reverse (ι Q m) = ι Q m :=
+by simp [reverse]
+
+@[simp] lemma reverse.commutes (r : R) :
+  reverse (algebra_map R (clifford_algebra Q) r) = algebra_map R _ r :=
+by simp [reverse]
+
+@[simp] lemma reverse.map_one  : reverse (1 : clifford_algebra Q) = 1 :=
+reverse.commutes 1
+
+@[simp] lemma reverse.map_mul (a b : clifford_algebra Q) :
+  reverse (a * b) = reverse b * reverse a :=
+by simp [reverse]
+
+@[simp] lemma reverse_comp_reverse :
+  reverse.comp reverse = (linear_map.id : _ →ₗ[R] clifford_algebra Q) :=
+begin
+  ext m,
+  simp only [linear_map.id_apply, linear_map.comp_apply],
+  induction m using clifford_algebra.induction,
+  -- simp can close these goals, but is slow
+  case h_grade0 : { rw [reverse.commutes, reverse.commutes] },
+  case h_grade1 : { rw [reverse_ι, reverse_ι] },
+  case h_mul : a b ha hb { rw [reverse.map_mul, reverse.map_mul, ha, hb], },
+  case h_add : a b ha hb { rw [reverse.map_add, reverse.map_add, ha, hb], },
+end
+
+@[simp] lemma reverse_involutive : function.involutive (reverse : _ → clifford_algebra Q) :=
+linear_map.congr_fun reverse_comp_reverse
+
+lemma reverse_comp_involute :
+  reverse.comp involute.to_linear_map =
+    (involute.to_linear_map.comp reverse : _ →ₗ[R] clifford_algebra Q) :=
+begin
+  ext,
+  simp only [linear_map.comp_apply, alg_hom.to_linear_map_apply],
+  induction x using clifford_algebra.induction,
+  case h_grade0 : { simp },
+  case h_grade1 : { simp },
+  case h_mul : a b ha hb { simp [ha, hb], },
+  case h_add : a b ha hb { simp [ha, hb], },
+end
+
+/-- `clifford_algebra.reverse` and `clifford_algebra.inverse` commute. Note that the composition
+is sometimes referred to as the "clifford conjugate". -/
+lemma reverse_involute_commute : function.commute (reverse : _ → clifford_algebra Q) involute :=
```

Comment:
Does it make sense to add a fully applied version of this statement, for easy (and short) `rw`s?

Reply:
Yes, good point.

Reply:
Done